staging/src/k8s.io/apiserver/pkg/registry/rest/create.go

```diff
@@ -145,21 +147,31 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx context.Context, obj runtime.
 
 // CheckGeneratedNameError checks whether an error that occurred creating a resource is due
 // to generation being unable to pick a valid name.
-func CheckGeneratedNameError(strategy RESTCreateStrategy, err error, obj runtime.Object) error {
+func CheckGeneratedNameError(ctx context.Context, strategy RESTCreateStrategy, err error, obj runtime.Object) error {
 	if !errors.IsAlreadyExists(err) {
 		return err
 	}
 
-	objectMeta, kind, kerr := objectMetaAndKind(strategy, obj)
+	objectMeta, gvk, kerr := objectMetaAndKind(strategy, obj)
 	if kerr != nil {
 		return kerr
 	}
 
 	if len(objectMeta.GetGenerateName()) == 0 {
+		// If we don't have a generated name, return the original error (AlreadyExists).
+		// When we're here, the user picked a name that is causing a conflict.
 		return err
 	}
 
-	return errors.NewServerTimeoutForKind(kind.GroupKind(), "POST", 0)
+	requestInfo, _ := genericapirequest.RequestInfoFrom(ctx)
```

Comment:
Is this ok to ignore? What should the fallback be in case the second returned value is false?

Reply:
in all expected cases, this will be populated, but no, it's not ok to ignore... requestInfo will be nil in that case and this code will panic

if there's no request info, I'd pass an empty `schema.GroupResource{}`, and only include the resource info in the message if the resource is non-empty

Reply:
Done